api/context.go

```diff
@@ -90,6 +96,27 @@ func (c context) WithContext(ctx gocontext.Context) Context {
 	return &c
 }
 
+func shouldSample(perc int) bool {
+	if perc >= 100 {
+		return true
+	}
+	if perc <= 0 {
+		return false
+	}
+	randomNum := rand.Intn(100)
+	return perc >= randomNum
+}
+
+func (c context) StartTrace(tracerName, spanName string, samplingPerc int) (Context, trace.Span) {
```

Comment:
I think it would be better if we split this into a StartTrace and StartSampledTrace

also why do we need both tracerName and span name ?

Reply:
tracer-name and span-name are different tags that otel sets on its own